src/main/java/com/gluonhq/substrate/target/LinuxLinkerFlags.java

```diff
@@ -0,0 +1,190 @@
+     */
+    private static final List<PkgInfo> LINK_DEPENDENCIES = List.of(
+        hardwired("-Wl,--no-whole-archive"),
+
+        activeOf(debian("gl", "libgl-dev"),
+                 fedora("gl", "mesa-libGL-devel")),
+        activeOf(debian("x11", "libx11-dev"),
+                 fedora("x11", "libX11-devel")),
+        activeOf(debian("gtk+-x11-3.0", "libgtk-3-dev"),
+                 fedora("gtk+-3.0", "gtk3-devel")),
+        activeOf(debian("freetype2", "libfreetype6-dev"),
+                 fedora("freetype2", "freetype-devel")),
+        activeOf(debian("pangoft2", "libpango1.0-dev"),
+                 fedora("pangoft2", "pango-devel")),
+
+        hardwired("-lgstreamer-lite"),
+
+        activeOf(debian("gthread-2.0", "libglib2.0-dev"),
+                 fedora("gthread-2.0", "glib2-devel")),
+
+        hardwired("-lstdc++"),
+        
+        activeOf(debian("zlib", "zlib1g-dev"),
+                 fedora("zlib", "zlib-devel")),
+
+        activeOf(debian("xtst", "libxtst-dev"),
+                 fedora("xtst", "libXtst-devel")),
+
+        // On fedora these require https://rpmfusion.org/
+        activeOf(debian("libavcodec", "libavcodec-dev"),
+                 fedora("libavcodec", "ffmpeg-devel")),
+        activeOf(debian("libavformat", "libavformat-dev"),
+                 fedora("libavformat", "ffmpeg-devel")),
+        activeOf(debian("libavutil", "libavutil-dev"),
+                 fedora("libavutil", "ffmpeg-devel")),
+
+        activeOf(debian("alsa", "libasound2-dev"),
+                 fedora("alsa", "alsa-lib-devel")),
+
+        hardwired("-lm"),
+
+        activeOf(debian("gmodule-no-export-2.0", "libglib2.0-dev"),
+                 fedora("gmodule-no-export-2.0", "glib2-devel"))
+    );
+
+    public List<String> getLinkerFlags() {
+        List<String> missingPackages = new ArrayList<>();
+
+        List<String> pkgFlags = LINK_DEPENDENCIES.stream()
+                .flatMap(pkg -> lookupPackageFlags(pkg, missingPackages).stream())
+                .collect(Collectors.toList());
+
+        if (!missingPackages.isEmpty()) {
+            printUpdateInstructionsAndFail(missingPackages);
+        }
+
+        Logger.logDebug("All flags: " + pkgFlags);
+        return pkgFlags;
+    }
+
+    /**
+     * Uses pkg-config to lookup linker flags for package.
+     * 
+     * If pkg-config fails, adds amendment instructions to missingPackages.
+     */
+    private List<String> lookupPackageFlags(PkgInfo pkgInfo, List<String> missingPackages) {
+        if (pkgInfo.hardwired != null) {
+            return List.of(pkgInfo.hardwired);
+        }
+
+        String pkgName = pkgInfo.pkgName;
+        ProcessRunner process = new ProcessRunner("/usr/bin/pkg-config", "--libs", pkgName);
+        try {
+            if (process.runProcess("Get config for " + pkgName) != 0) {
+                missingPackages.add(pkgInfo.installName + " (for pkgConfig " + pkgName + ")");
+                return List.of();
```

Comment:
In this case, should there be some message to the caller that something went wrong?

Reply:
There is. In the caller (handed off to printUpdateInstructionsAndFail)
The information is summarized, so you get a list of all packages to install, instead of doing repeated failed builds, adding a missing package each time.

I think it might even be worth calling this code from the checkLinker method, so you get a list of missing OS packages to install, before having waited for the slow compilation step.

Reply:
And btw, I think the build timed out because you had a build running at the same time.
My plan is to squash the commits and thus trigger a rebuild during the weekend.

Reply:
@johanvos do I need to do something more to get this landed?

Reply:
@abhinayagarwal is doing a review.
In general, I think this is a good step.
However, there is a deeper issue that is unsolved yet, and that has to tackled one day too. The JavaFX SDK will create 3 versions of libavplugin.so: libavplugin-54.so, libavplugin-56.so and libavplugin-57.so. There versions each depend on specific corresponding versions of libavcodec-XX.so. At runtime, if libavplugin-57.so can be loaded (e.g. if libavcodec-57.so is on the system), that one is used. Otherwise, lower versions are tried.
We don't have this logic yet in our static builds. But I think this should be dealt with in another issue.

Reply:
@jskov but in general, this looks very good!

Reply:
@johanvos thanks! I will take time to address the last comments in the weekend.

Reply:
@johanvos I have pushed a new version which addressed the comments (except for @jperedadnr 's about exception handling).

The Travis setup has some problems with stability.
My first push failed, I believe because another PR submission was building at the same time.
I pushed a whitespace-only change and the new build was allowed to run alone and succeeded.

Cheers!

Reply:
@jskov You say you have pushed new changes? I only see one commit from 21 days ago. Can you verify?

Reply:
@jperedadnr 
Yes the current diff reflects the changes I made Sunday (e.g.  "-Wl,--no-whole-archive" added in LinuxTargetConfiguration.java).
I flattened the changes into a single commit - and that is probably the original one.

Sorry, did not realize this would happen. Somewhat hides the changes I made along the way.
But that was how quarkus people wanted PRs, so I just repeated here. My bad.

Reply:
We usually squash at the end right before merging, otherwise it is hard to see the changes in the revision process.

Reply:
Which is the sensible way to do it :) Also what I do at work.
My apologies.